internal/test/e2e/otelglobal/main.go

```diff
@@ -31,11 +31,19 @@ func innerFunction(ctx context.Context) {
 	span.SetAttributes(attribute.String("inner.key", "inner.value"))
 }
 
+func setNameOverride(ctx context.Context) {
+	_, span := tracer.Start(ctx, "SetNameOriginal")
+	defer span.End()
+
+	span.SetName("SetNameOverride")
+}
+
 func createMainSpan(ctx context.Context) {
 	ctx, span := tracer.Start(ctx, "parent")
 	defer span.End()
 
 	innerFunction(ctx)
+	setNameOverride(ctx)
```

Comment:
I think this structure of one parent span with two children will make it more complicated for the `verify.bats`. It is possible, but maybe worth considering calling `SetName` on one of the two spans without adding a third one.

Reply:
that makes sense. i thought about doing that, but then opted to make a new span to keep the components i'm testing more isolated. i'll make the other adjustments, then look into simplifying this test/validation more.